File: java/tests/sorting/heap_sort/HeapSortTest.java
package sorting;
import java.util.List;
import java.util.Arrays;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class HeapSortTest {

    @Test
    public void testSortAscBasic() {
        List<Integer> input = Arrays.asList(4, 5, 5, 10, 10, 8, 2, 100, 50, 30, 1);
        List<Integer> expected = Arrays.asList(1, 2, 4, 5, 5, 8, 10, 10, 30, 50, 100);

        List<Integer> result = HeapSort.sortAsc(input);
        assertEquals(expected, result);
    }

    @Test
    public void testSortDescBasic() {
        List<Integer> input = Arrays.asList(4, 5, 5, 10, 10, 8, 2, 100, 50, 30, 1);
        List<Integer> expected = Arrays.asList(100, 50, 30, 10, 10, 8, 5, 5, 4, 2, 1);

        List<Integer> result = HeapSort.sortDesc(input);
        assertEquals(expected, result);
    }

    @Test
    public void testEmptyList() {
        List<Integer> input = List.of();
        List<Integer> expected = List.of();

        assertEquals(expected, HeapSort.sortAsc(input));
        assertEquals(expected, HeapSort.sortDesc(input));
    }

    @Test
    public void testSingleElementList() {
        List<Integer> input = List.of(42);
        List<Integer> expectedAsc = List.of(42);
        List<Integer> expectedDesc = List.of(42);

        assertEquals(expectedAsc, HeapSort.sortAsc(input));
        assertEquals(expectedDesc, HeapSort.sortDesc(input));
    }

    @Test
    public void testAllEqualElements() {
        List<Integer> input = Arrays.asList(7, 7, 7, 7, 7);
        List<Integer> expected = Arrays.asList(7, 7, 7, 7, 7);

        assertEquals(expected, HeapSort.sortAsc(input));
        assertEquals(expected, HeapSort.sortDesc(input));
    }
}

File: csharp/tests/sorting/heap_sort/HeapSortTest.cs
using Algorithms.Sorting;
using System;
using System.Collections.Generic;
using Xunit;

namespace Algorithms.Tests.Sorting;

public class HeapSortTest
{
    [Fact]
    public void TestSortAscBasic()
    {
        var input = new List<int> { 4, 5, 5, 10, 10, 8, 2, 100, 50, 30, 1 };
        var expected = new List<int> { 1, 2, 4, 5, 5, 8, 10, 10, 30, 50, 100 };

        var result = HeapSort.SortAsc(input);

        Assert.Equal(expected, result);
    }

    [Fact]
    public void TestSortDescBasic()
    {
        var input = new List<int> { 4, 5, 5, 10, 10, 8, 2, 100, 50, 30, 1 };
        var expected = new List<int> { 100, 50, 30, 10, 10, 8, 5, 5, 4, 2, 1 };

        var result = HeapSort.SortDesc(input);

        Assert.Equal(expected, result);
    }

    [Fact]
    public void TestEmptyList()
    {
        var input = new List<int>();
        var expected = new List<int>();

        Assert.Equal(expected, HeapSort.SortAsc(input));
        Assert.Equal(expected, HeapSort.SortDesc(input));
    }

    [Fact]
    public void TestSingleElementList()
    {
        var input = new List<int> { 42 };
        var expectedAsc = new List<int> { 42 };
        var expectedDesc = new List<int> { 42 };

        Assert.Equal(expectedAsc, HeapSort.SortAsc(input));
        Assert.Equal(expectedDesc, HeapSort.SortDesc(input));
    }

    [Fact]
    public void TestAllEqualElements()
    {
        var input = new List<int> { 7, 7, 7, 7, 7 };
        var expected = new List<int> { 7, 7, 7, 7, 7 };

        Assert.Equal(expected, HeapSort.SortAsc(input));
        Assert.Equal(expected, HeapSort.SortDesc(input));
    }
}
